gh-132661: Document t-strings and `templatelib` (#135229)
python/cpython: Doc/library/ast.rst

   * ``conversion`` is an integer:

     * -1: no formatting

>      * 115 (``ord('s')``): ``!s`` string formatting
>      * 114 (``ord('r')``): ``!r`` repr formatting
>      * 97 (``ord('a')``): ``!a`` ASCII formatting

Reviewer: ```suggestion
     * 97: ``!a`` :func:`ASCII <ascii>` formatting (``ord('a')``)
     * 114: ``!r`` :func:`repr` formatting (``ord('r')``)
     * 115: ``!s`` :func:`string <str>` formatting (``ord('s')``)
```

                                Constant(value='.3')]))]))

> .. class:: TemplateStr(values)
> 
>    A t-string, comprising a series of :class:`Interpolation` and :class:`Constant`
>    nodes.

Reviewer: Add a note on AST node ordering
```suggestion
.. class:: TemplateStr(values, /)

   Node representing a template string literal, comprising a series of
   :class:`Interpolation` and :class:`Constant` nodes.
   These nodes may be any order, and do not need to be interleaved.
```

   .. doctest::

        >>> print(ast.dump(ast.parse('t"{name} finished {place:ordinal}"', mode='eval'), indent=4))
        Expression(
            body=TemplateStr(
                values=[
                    Interpolation(
                        value=Name(id='name'),
                        str='name',
                        conversion=-1),
                    Constant(value=' finished '),
                    Interpolation(
                        value=Name(id='place'),
                        str='place',
                        conversion=-1,
                        format_spec=JoinedStr(
                            values=[
                                Constant(value='ordinal')]))]))

   .. versionadded:: 3.14

> .. class:: Interpolation(value, str, conversion, format_spec)

Reviewer: ```suggestion
.. class:: Interpolation(value, str, conversion, format_spec=None)
```

>    Node representing a single interpolation field in a t-string.

Reviewer: ```suggestion
   Node representing a single interpolation field in a template string literal.
```

   * ``value`` is any expression node (such as a literal, a variable, or a
     function call).
   * ``str`` is a constant containing the text of the interpolation expression.
   * ``conversion`` is an integer:

     * -1: no conversion
     * 115: ``!s`` string conversion
     * 114: ``!r`` repr conversion

>      * 97: ``!a`` ascii conversion

Reviewer 2: ```suggestion
     * 97: ``!a`` ASCII conversion
```

>      * 115: ``!s`` string conversion
>      * 114: ``!r`` repr conversion
>      * 97: ``!a`` ascii conversion

Reviewer 3: Should we reveal how they were chosen?
```suggestion
     * 115 (``ord('s')``): ``!s`` string conversion
     * 114 (``ord('r')``): ``!r`` repr conversion
     * 97 (``ord('a')``): ``!a`` ascii conversion
```

Author: That's a nice addition; thanks!

>      * 115: ``!s`` string conversion
>      * 114: ``!r`` repr conversion
>      * 97: ``!a`` ascii conversion

Reviewer: ```suggestion
     * 97: ``!a`` :func:`ASCII <ascii>` conversion (``ord('a')``)
     * 114: ``!r`` :func:`repr` conversion (``ord('r')``)
     * 115: ``!s`` :func:`string <str>` conversion (``ord('s')``)
```
